src/azstoragetorch/datasets.py

```diff
@@ -370,7 +370,7 @@ def __iter__(self) -> Iterator[_TransformOutputType_co]:
         worker_info = torch.utils.data.get_worker_info()
         for i, blob in enumerate(self._blobs):
             if self._should_yield_from_worker_shard(worker_info, i):
-                yield self._transform(blob)
+                yield self._transform(blob) # type: ignore[misc]
```

Comment:
Can we remove the type ignores now that we are casting the default transform callable?